include/aws/common/common.h

```diff
@@ -41,6 +41,20 @@
                                     abort();                                                                    \
                                }                                                                                \
 
+/**
+ * Define a function contract to encapsulate the check of a postcondition with
+ * the return value of a function. The cost in prod is 0, because when the
+ * DEBUG mode is turn off the condition is not even evaluated. This keep the
+ * assert in the same line for debuggability.
+ */
+#ifdef NDEBUG
```

Comment:
You don't need the ndebug check anymore, since the assert will already handle that.